smwe_rom/src/compression/lc_lz2.rs

```diff
@@ -1,40 +1,32 @@
-use std::convert::TryFrom;
-
-use num_enum::TryFromPrimitive;
-
 use crate::error::{DecompressionError, LcLz2Error};
 
-#[repr(u8)]
-#[derive(Copy, Clone, Debug, TryFromPrimitive)]
-enum Command {
-    /// Followed by (L+1) bytes of data
-    DirectCopy     = 0b000,
+/// Followed by (L+1) bytes of data
+const DIRECT_COPY: u8 = 0b000;
 
-    /// Followed by one byte to be repeated (L+1) times
-    ByteFill       = 0b001,
+/// Followed by one byte to be repeated (L+1) times
+const BYTE_FILL: u8 = 0b001;
 
-    /// Followed by two bytes. Output first byte, then second, then first,
-    /// then second, etc. until (L+1) bytes has been outputted
-    WordFill       = 0b010,
+/// Followed by two bytes. Output first byte, then second, then first,
+/// then second, etc. until (L+1) bytes has been outputted
+const WORD_FILL: u8 = 0b010;
 
-    /// Followed by one byte to be repeated (L+1) times, but the byte is
-    /// increased by 1 after each write
-    IncreasingFill = 0b011,
+/// Followed by one byte to be repeated (L+1) times, but the byte is
+/// increased by 1 after each write
+const INCREASING_FILL: u8 = 0b011;
 
-    /// Followed by two bytes (ABCD byte order) containing address (in the
-    /// output buffer) to copy (L+1) bytes from
-    Repeat         = 0b100,
+/// Followed by two bytes (ABCD byte order) containing address (in the
+/// output buffer) to copy (L+1) bytes from
+const REPEAT: u8 = 0b100;
 
-    /// This command has got a two-byte header:
-    /// ```text
-    /// 111CCCLL LLLLLLLL
-    /// CCC:        Real command
-    /// LLLLLLLLLL: Length
-    /// ```
-    LongLength     = 0b111,
-}
+/// This command has got a two-byte header:
+/// ```text
+/// 111CCCLL LLLLLLLL
+/// CCC:        Real command
+/// LLLLLLLLLL: Length
+/// ```
+const LONG_LENGTH: u8 = 0b111;
 
-pub fn decompress(input: &[u8]) -> Result<Vec<u8>, DecompressionError> {
+pub fn decompress(input: &[u8], revised_gfx: bool) -> Result<Vec<u8>, DecompressionError> {
```

Comment:
I'd rename `revised_gfx` to something more generic like `little_endian_in_repeat`, since decompression code doesn't necessarily have to deal with graphics, even though GFX might be the only thing LZ2 is used for in vanilla SMW.